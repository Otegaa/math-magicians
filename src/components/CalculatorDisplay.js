import React, { useState } from 'react';

const CalculatorDisplay = () => {
  const [value, setValue] = useState([0]);

  const handleClick = (num) => (value[0] === 0 ? setValue(num) : setValue([...value, num]));

  return (
    <div className="container">
      <div className="display-calc line">
        <span>{value}</span>
      </div>

      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        AC
      </button>
      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        +/-
      </button>
      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        %
      </button>
      <button
        type="button"
        className="btn-orange"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        &#xf7;
      </button>

      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        7
      </button>
      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        8
      </button>
      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        9
      </button>
      <button
        type="button"
        className="btn-orange"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        x
      </button>

      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        4
      </button>
      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        5
      </button>
      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        6
      </button>
      <button
        type="button"
        className="btn-orange"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        -
      </button>

      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        1
      </button>
      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        2
      </button>
      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        3
      </button>
      <button
        type="button"
        className="btn-orange"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        +
      </button>

      <button
        type="button"
        className="btn-white btn-zero"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        0
      </button>
      <button
        type="button"
        className="btn-white"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        .
      </button>
      <button
        type="button"
        className="btn-orange"
        onClick={(e) => handleClick(e.target.textContent)}
      >
        =
      </button>
    </div>
  );
};

export default CalculatorDisplay;
