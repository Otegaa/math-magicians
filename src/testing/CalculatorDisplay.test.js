import '@testing-library/jest-dom';
import { render, fireEvent } from '@testing-library/react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import CalculatorDisplay from '../components/CalculatorDisplay';

describe('Calculator', () => {
  it('Should render correctly', () => {
    const { getByText } = render(<CalculatorDisplay />);
    expect(getByText('AC')).toBeInTheDocument();
    expect(getByText('+/-')).toBeInTheDocument();
    expect(getByText('%')).toBeInTheDocument();
    expect(getByText('รท')).toBeInTheDocument();
    expect(getByText('x')).toBeInTheDocument();
    expect(getByText('-')).toBeInTheDocument();
    expect(getByText('+')).toBeInTheDocument();
    expect(getByText('.')).toBeInTheDocument();
    expect(getByText('=')).toBeInTheDocument();
  });
});

test('should change symbol when the "+/-" button is clicked', () => {
  const { getByText } = render(<CalculatorDisplay />);
  fireEvent.click(getByText('9'));
  fireEvent.click(getByText('+/-'));
  expect(getByText('-9')).toBeInTheDocument();
});

it('renders correctly', () => {
  const tree = renderer.create(<CalculatorDisplay />).toJSON();
  expect(tree).toMatchSnapshot();
});
