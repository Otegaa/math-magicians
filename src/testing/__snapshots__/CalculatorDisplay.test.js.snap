// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="display-calc line"
  >
    <span>
      0
    </span>
  </div>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="btn-orange"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="btn-orange"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="btn-orange"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="btn-orange"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="btn-white btn-zero"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="btn-white"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="btn-orange"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
